### PR TITLE
return 0 for getDelegateRewards if voter didn't vote in the screening stage

### PR DESCRIPTION
**Pull request by MikeHathaway**
# Description of change
## High level
* return 0 for `getDelegateRewards` if voter didn't vote in the screening stage

# Description of bug or vulnerability and solution
* `getDelegateReward` returned incorrect values if called when a voter didn't vote in the screening stage.
* https://github.com/Fixed-Point-Solutions/ajna-PositionManager-RewardsManager/issues/24

### CODE REVIEW

**Review by grandizzy — COMMENTED** [review]
shouldn't this check be done inside `_getDelegateReward` function so to be applied for all scenarios where it is called?

File: src/grants/GrantFund.sol [hunk 1]
```diff
@@ -1099,6 +1099,8 @@ contract GrantFund is IGrantFund, Storage, ReentrancyGuard {
         DistributionPeriod storage currentDistribution = _distributions[distributionId_];
         VoterInfo          storage voter               = _voterInfo[distributionId_][voter_];
 
+        if (voter.screeningVotesCast == 0) return 0;
```
Comment (grandizzy) [inline src/grants/GrantFund.sol:1102]: I personally prefer using
```Solidity
if (voter.screeningVotesCast != 0) {
  rewards_ = 
}
```

**Comment by MikeHathaway** [comment]
> shouldn't this check be done inside `_getDelegateReward` function so to be applied for all scenarios where it is called?

Did it this way for a couple of reasons - this would be the minimal amount of changes, and also, the claim approach should revert instead of returning 0, so we would have to create branching logic to handle claim vs get calls to `_getDelegateReward`

**Comment by grandizzy** [comment]
> > shouldn't this check be done inside `_getDelegateReward` function so to be applied for all scenarios where it is called?
> 
> Did it this way for a couple of reasons - this would be the minimal amount of changes, and also, the claim approach should revert instead of returning 0, so we would have to create branching logic to handle claim vs get calls to `_getDelegateReward`

makes sense, thanks

**Review by EdNoepel — APPROVED** [review]
LGTM

File: src/grants/GrantFund.sol (hunk 1 above)
Comment (ith-harvey) [reply src/grants/GrantFund.sol:1102]: Although I'm not opinionated about this change, it's a pattern we have been following in the contracts repo and therefore I agree with @grandizzy .

**Review by ith-harvey — APPROVED** [review]
LGTM

**Review by prat-gpt — APPROVED** [review]
LGTM

**Pull request merged**